server/handlers/chat_handler.js

```diff
@@ -1,16 +1,51 @@
+import path from 'path';
+import { HNSWLib } from "langchain/vectorstores/hnswlib";
+import { RetrievalQAChain } from 'langchain/chains';
+import { OpenAIEmbeddings } from "langchain/embeddings/openai";
+import { ContextualCompressionRetriever } from "langchain/retrievers/contextual_compression";
+import { LLMChainExtractor } from "langchain/retrievers/document_compressors/chain_extract";
+
 import { MODEL_STORES, enabledModel } from '../config/model_store_constants.js';
+import { getFileLoader } from '../utils/documentLoader.js';
 
 class ChatService {
   constructor () {
-    this.model = new MODEL_STORES[enabledModel]
+    this.modelService = new MODEL_STORES[enabledModel]
   }
 
   async startChat(data) {
     const { body: { userInput } } = data;
-    const response = await this.model.call(userInput);
-
+    const response = await this.modelService.call(userInput);
     return response;
   }
+
+  async ingestFile(data) {
```

Comment:
I could see the logic of this function going in the `/services/openai` file since it is dependent on OpenAi. And in this file writing something like `this.modelService.callWithDocument(docs)`

Reply:
moved it 🧥